tasks/Module 5/task_2.py

```diff
@@ -0,0 +1,231 @@
+            hero.position = (aligned_init_pos + 1, initial_position[1])
+        elif direction in ("west", "l"):
+            aligned_init_pos = (
+                len(self.field[0]) if initial_position[1] == 0 else initial_position[1]
+            )
+            cell_content = self.field[initial_position[0]][aligned_init_pos - 1]
+            self.field[initial_position[0]][aligned_init_pos - 1] = "H"
+            hero.position = (initial_position[0], aligned_init_pos - 1)
+        elif direction in ("east", "r"):
+            aligned_init_pos = (
+                -1
+                if initial_position[1] == len(self.field[0]) - 1
+                else initial_position[1]
+            )
+            cell_content = self.field[initial_position[0]][aligned_init_pos + 1]
+            self.field[initial_position[0]][aligned_init_pos + 1] = "H"
+            hero.position = (initial_position[0], aligned_init_pos + 1)
+        # clear initial hero position
+        self.field[initial_position[0]][initial_position[1]] = " "
+        # print('position after move', hero.position)
+        return cell_content
+
+    def __str__(self):
+        """Draw field."""
+        to_print = ""
+        for _, row in enumerate(self.field):
+            to_print += "+---" * self.columns + "+\n"
+            to_print += (
+                "| "
+                + " | ".join(map(lambda x: "X" if isinstance(x, Item) else x, row))
+                + " |\n"
+            )
+        to_print += "+---" * self.columns + "+\n"
+        return to_print
+
+
+class Game:
+    """Main Game class."""
+
+    def __init__(self, columns, rows, hero):
+        """Init Game instance."""
+        self.field = Field(columns, rows)
+        self.field.place_items()
+        self.hero = hero
+        self.field.set_hero(self.hero)
+        self.debug = False
+
+    def __handle_responce(self, responce):
+        if responce in ("stats", "s"):
+            self.hero.get_stats()
+            _ = input()
+        if responce.startswith("drop"):
+            print(self.hero.inv)
+            responce = input("What to drop? ").rstrip()
+            self.hero.inv.remove_item(responce)
+        if responce in MOVES:
+            self.hero.stamina -= (
+                1 if self.hero.inv.get_weight() < LIGHT_WEIGHT_THRESHOLD else 4
+            )
+            cell_content = self.field.move_hero(self.hero, responce)
+            if self.debug:
+                print(cell_content, cell_content.__class__)
+            if self.debug:
+                print(self.hero.stamina)
+            if isinstance(cell_content, GameItem):
+                if cell_content.is_food():
+                    self.hero.heal(cell_content)
+                    return
+                print(f"New item found! It is a {cell_content.name}")
+                self.hero.inv.add_item(cell_content)
+                if self.hero.inv.check_weight() > 2:
+                    self.hero.can_move = False
```

Comment:
Do you have to set this property here? All properties required to check if hero can move are stored in `Hero` class anyway. Actually, do you even need to have `can_move` property?

Reply:
I agree we can go without this attribute, but instead of it i should every time calculate if self.hero.inv.check_weight() > 2

IMHO it is better reflects state of hero and code is more understandable. 
But if you deem i should remove it - no problem i will remove.

Reply:
What concerns me most here is the fact that `can_move` field is modified inside the `Hero` class and in the `Game` class. This way you can by mistake introduce inconsistent state in your code. Imagine a bug that leads to setting `can_move` to `False` when inventory is almost empty.

> I agree we can go without this attribute, but instead of it i should every time calculate if

You are right here and the most pythonic way to deal with such scenarios is to make `can_move` a property:
```python
class Hero:
    ...
    @property
    def can_move(self):
        return self.inv.check_weight() > 2
```
Maybe you will find it understandable, it is up to you.

Reply:
cool!
Thank you :)